#!/bin/sh /usr/share/dpatch/dpatch-run
## rss.dpatch by Torsten Landschoff <torsten@pulsar.galaxy>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Blank-ink-only patch by Richard Spencer-Smith

@DPATCH@
Index: hplip-3.12.4/prnt/hpijs/breaks_open.cpp
===================================================================
--- hplip-3.12.4.orig/prnt/hpijs/breaks_open.cpp	2012-04-10 18:39:32.000000000 +1000
+++ hplip-3.12.4/prnt/hpijs/breaks_open.cpp	2012-05-26 10:21:29.000000000 +1000
@@ -1078,7 +1078,10 @@
 
 {
 0,  0,  255,    0,
-0,  1,  255,    0,
+// 2001.06.16 mrb: changed the value to 0. With the original "1"
+//                 the printer sometimes prints empty lines...
+//0,	1,	255,	0,
+0,	 0,  255,    0,
 0,  3,  255,    0,
 0,  5,  255,    0,
 0,  7,  255,    0,
Index: hplip-3.12.4/prnt/hpijs/dj660_maps.cpp
===================================================================
--- hplip-3.12.4.orig/prnt/hpijs/dj660_maps.cpp	2012-04-10 18:39:32.000000000 +1000
+++ hplip-3.12.4/prnt/hpijs/dj660_maps.cpp	2012-05-26 10:21:29.000000000 +1000
@@ -38,7 +38,10 @@
 //const
 uint32_t ulMapDJ660_CCM_KCMY[ 9 * 9 * 9 ]=
     {
-        422720226UL, 142187374UL, 94354238UL, 62442785UL, 45668877UL, 27648516UL, 8972032UL, 7335680UL, 5895424UL,
+// 2001.06.14 mrb: Replaced the first entry so the printer will
+// use only the black pen when he should print real black.
+//		422720226UL, 142187374UL, 94354238UL, 62442785UL, 45668877UL, 27648516UL, 8972032UL, 7335680UL, 5895424UL, 
+ 		226UL, 142187374UL, 94354238UL, 62442785UL, 45668877UL, 27648516UL, 8972032UL, 7335680UL, 5895424UL,
 2519110233UL, 807846483UL, 391693109UL, 260029471UL, 143570703UL, 42776580UL, 8501760UL, 7782400UL, 6539520UL,
 3072173868UL, 1881129514UL, 841465387UL, 524792087UL, 358391818UL, 174364420UL, 40408577UL, 6920704UL, 6530560UL,
 3407591425UL, 2551891200UL, 1612695808UL, 824489479UL, 506699268UL, 306153730UL, 105283329UL, 5144320UL, 5737216UL,
Index: hplip-3.12.4/prnt/hpijs/dj690_maps.cpp
===================================================================
--- hplip-3.12.4.orig/prnt/hpijs/dj690_maps.cpp	2012-04-10 18:39:32.000000000 +1000
+++ hplip-3.12.4/prnt/hpijs/dj690_maps.cpp	2012-05-26 10:21:29.000000000 +1000
@@ -38,7 +38,10 @@
 uint32_t ulMapDJ690_CMYK[ 9 * 9 * 9 ]=
 // DJ690_CMYK
     {
- 337848575UL,   37838470UL,  155209268UL,  137905152UL,   36454656UL,    2310144UL,
+// 2001.06.14 mrb: Replaced the first entry so the printer will
+// use only the black pen when he should print real black.
+// 337848575UL,   37838470UL,  155209268UL,  137905152UL,   36454656UL,    2310144UL, 
+ 226UL,   37838470UL,  155209268UL,  137905152UL,   36454656UL,    2310144UL,
    2179072UL,    2051072UL,    1733120UL, 2013280119UL,  385897321UL,  303379741UL,
  270610688UL,  102772224UL,    1844992UL,    1911552UL,    1847296UL,    1784832UL,
 2080387626UL, 1157643294UL,  452996881UL,  335555072UL,  202384384UL,   50991872UL,
Index: hplip-3.12.4/prnt/hpijs/dj895_maps.cpp
===================================================================
--- hplip-3.12.4.orig/prnt/hpijs/dj895_maps.cpp	2012-04-10 18:39:32.000000000 +1000
+++ hplip-3.12.4/prnt/hpijs/dj895_maps.cpp	2012-05-26 10:21:29.000000000 +1000
@@ -37,7 +37,10 @@
 
 uint32_t ulMapDJ895_KCMY[ 9*9*9 ]=
 {
-1819176703UL, 79494215UL, 112983330UL, 79035667UL, 78183948UL, 60489224UL, 59571718UL, 42007811UL, 7405312UL,
+// 2001.06.14 mrb: Replaced the first entry so the printer will
+// use only the black pen when he should print real black.
+//1819176703UL, 79494215UL, 112983330UL, 79035667UL, 78183948UL, 60489224UL, 59571718UL, 42007811UL, 7405312UL, 
+226UL, 79494215UL, 112983330UL, 79035667UL, 78183948UL, 60489224UL, 59571718UL, 42007811UL, 7405312UL,
 4230922555UL, 655358293UL, 176553263UL, 128253208UL, 61209869UL, 60751368UL, 26606852UL, 9305345UL, 8650496UL,
 4248940299UL, 3776766759UL, 756415798UL, 308083743UL, 142671120UL, 59440646UL, 9239553UL, 9109248UL, 8779520UL,
 4248086274UL, 4214276619UL, 2786390812UL, 942206231UL, 491516427UL, 224131588UL, 6616320UL, 7399168UL, 7857664UL,
Index: hplip-3.12.4/prnt/hpijs/dj895_maps2.cpp
===================================================================
--- hplip-3.12.4.orig/prnt/hpijs/dj895_maps2.cpp	2012-04-10 18:39:32.000000000 +1000
+++ hplip-3.12.4/prnt/hpijs/dj895_maps2.cpp	2012-05-26 10:21:29.000000000 +1000
@@ -163,7 +163,10 @@
 
 uint32_t ulMapDJ895_Binary_KCMY[ 9*9*9 ]=
 {
-1819176648UL,   64158794UL,   80805158UL,   80215318UL,   62389519UL,   61406730UL,
+// 2001.06.14 mrb: Replaced the first entry so the printer will
+// use only the black pen when he should print real black.
+//1819176648UL,   64158794UL,   80805158UL,   80215318UL,   62389519UL,   61406730UL, 
+226UL,   64158794UL,   80805158UL,   80215318UL,   62389519UL,   61406730UL,
   60292615UL,   42401028UL,    7601920UL, 4230922301UL,  571406681UL,  160889651UL,
  113048859UL,   62717200UL,   45415944UL,   27917573UL,   10419457UL,    9238272UL,
 4248940557UL, 3743081259UL,  790296119UL,  325515810UL,  127007505UL,   27196678UL,
Index: hplip-3.12.4/prnt/hpijs/dj970_maps.cpp
===================================================================
--- hplip-3.12.4.orig/prnt/hpijs/dj970_maps.cpp	2012-04-10 18:39:32.000000000 +1000
+++ hplip-3.12.4/prnt/hpijs/dj970_maps.cpp	2012-05-26 10:21:29.000000000 +1000
@@ -127,7 +127,11 @@
 
 uint32_t ulMapDJ970_KCMY[ 9*9*9 ]=
 {
- 673723135UL,   26007371UL,    9756717UL,   10020638UL,   10022677UL,   10153743UL, 
+// 2003.06.05 rss:  Replaced the first entry so the printer will
+// (hopefully) use only the black pen when he should print real black.
+// Please report as a bug if it doesn't, or something worse happens.
+// 673723135UL,   26007371UL,    9756717UL,   10020638UL,   10022677UL,   10153743UL, 
+       226UL,   26007371UL,    9756717UL,   10020638UL,   10022677UL,   10153743UL, 
   10285066UL,   10416133UL,    9169408UL, 3405886809UL,      59743UL,   23980849UL, 
    9366812UL,   10153744UL,   10285066UL,   10153990UL,   10087424UL,    9169408UL, 
 3858807091UL, 2197868343UL,      60220UL,    4714015UL,    7270155UL,    8646656UL, 
Index: hplip-3.12.4/prnt/hpijs/dj970_maps3.cpp
===================================================================
--- hplip-3.12.4.orig/prnt/hpijs/dj970_maps3.cpp	2012-04-10 18:39:32.000000000 +1000
+++ hplip-3.12.4/prnt/hpijs/dj970_maps3.cpp	2012-05-26 10:21:29.000000000 +1000
@@ -37,7 +37,8 @@
 
 uint32_t ulMapDJ970_Draft_KCMY[ 9*9*9 ]=
 {
-1802205183UL,   22740362UL,    5832524UL,    5635892UL,    5570340UL,    5308188UL,
+//1802205183UL,   22740362UL,    5832524UL,    5635892UL,    5570340UL,    5308188UL,
+       226UL,   22740362UL,    5832524UL,    5635892UL,    5570340UL,    5308188UL,
    5308180UL,    5504778UL,    5504768UL, 4278239136UL,      65446UL,    4194132UL,
    5308210UL,    5373728UL,    5308182UL,    5504784UL,    5504776UL,    5504768UL,
 4278498896UL, 1862336354UL,      65388UL,    2621242UL,    4128540UL,    4718348UL,
